### DISCUSSION TITLE
Is there a known max. bandwidth / speed limit for this to work?

### DISCUSSION THREAD

**Discussion by TipsTricksMore** [Ysurac/openmptcprouter#2927] [Q&A]
Hi there,

first of all a huge THANK YOU for developing this great project @Ysurac 

I'm running a small video company in Germany and bandwidth is pretty critical for our daily work. We're uploading a lot of large files to off-site servers, clients, etc.
Unfortunately, the highest-paid service at my location 'only' offers 1.000 Mbit Down and 50 Mbit Up... 1.000 Down is acceptable but only 50 Up slows us down during work and is just exhausting...

However, there could be a solution to that problem!
While my ISP (yes it's the only one in my region) offers not more than these speeds I would be able to just get a second or third line and thus double or triple my Up- and Downlink over these multiple lines. 

So my plan would be to just sign up for a second line for now and do bonding to achieve approx. double the Up and Downlink.

I already did a lot of research on finding router hardware that can run openMPTCPRouter at my local end with multiple 2.5Gig Ports to achieve those speeds of approx: 2.000 Down and 100 Up. 
Our main switches are all 10Gig but this 2.5Gig router would only be for the initial connection to the "outside" / ISP's so it should be fine for now.

I'm used to working with vServers, Debian, ubuntu etc... and we're using some for our work. 
I know that the bandwidth of the vServer could be the limiting factor for this project. Even though we would be fine by only having the 100 Upload, the techie person inside of me obviously wants to use the full 2.000 Download as well.. ;)

All our recent vServers had a shared 1Gig Up- Downlink which would be not sufficient enough... So I began searching for other hosting services in my area and found one that offered vServers with a shared 20Gig Up- and Downlink for a reasonable price of only 10 EUR / month.

So I booked a debian10 server at their site (AMD EPYC 2, 2 dedicated cores, 8 GB RAM, 40GB NVMe, 2x 10 Gbit/s shared, unlimited traffic) and did some testing.

A simple speedtest to the nearest server showed up to 7.000 Up- and Downlink and with iperf3 I was able to fully max out my local line using 1 Gbit Down...I ran an automated script for a few days to periodically test the bandwidth and these results were constant. 

**QUESTIONS:**
- That's great and should mean, that this vServer probably handles 2.000 Down and 100 Up using bonding easily right? 
I have no experience if there's any limit in speeds using openMPTCPRouter but would assume there isn't one at this level..?!
I already installed the VPS side of openMPTCPRouter without problems.

- My router hardware will take some time until it arrives but should also be sufficient enough to handle those speeds..!?
Thats the router hardware I opted for: https://de.aliexpress.com/item/1005004359957893.html (N5105 i225-V variant, I'll add 8GB RAM and let openMPTCPRouter boot via usb)
According to this vid, the throughput is around 2.3Gbit per port: https://youtu.be/kQamMghox30

I would be pleased to get some feedback on this :)

Thanks a lot, you all! 

Best
Leo

**Comment by Ysurac** [comment]
There is no speed limits, it was already tested with 10Gb/s aggregated speed.
The limit is often connection and hardware speed. Aggregation is using some CPU, and even more with the encryption (AES-NI hardware support help on that).

**Reply by TipsTricksMore** [reply]
Thanks for your fast reply!

You got any experience with the hardware I choose? Does the N5105 deliver enough CPU power?

I have no experience in how hungry it is and would highly appreciate some values to compare.

Thank you!
